cmd/kube-proxy/app/server.go

```diff
@@ -704,3 +699,17 @@ func getConntrackMax(config kubeproxyconfig.KubeProxyConntrackConfiguration) (in
 	}
 	return 0, nil
 }
+
+// CleanUpAndExit remove iptables rules and exit if success return nil
+func (s *ProxyServer) CleanUpAndExit() error {
+	// remove iptables rules and exit
+	if s.CleanupAndExit {
```

Comment:
do we still need this field and test now?

Reply:
ping @vllry @andrewsykim

Reply:
I have noticed that ``CleanupAndExit`` option, are this different?
https://github.com/kubernetes/kubernetes/blob/716344fd7d68f92d9a16280be1feae0ffe122bb4/cmd/kube-proxy/app/server.go#L97-L103
https://github.com/kubernetes/kubernetes/blob/716344fd7d68f92d9a16280be1feae0ffe122bb4/cmd/kube-proxy/app/server.go#L500
https://github.com/kubernetes/kubernetes/blob/716344fd7d68f92d9a16280be1feae0ffe122bb4/cmd/kube-proxy/app/server_others.go#L99-L106

Reply:
thockin is right, that if-statement can be removed, as the function is only called when CleanupAndExit is set.

CleanupAndExit / CleanUpAndExit are the same thing, there just appears to be some internal inconsistencies in variable names.

Reply:
Updated
Thanks